src/decaf.cpp

```diff
@@ -78,6 +78,11 @@ namespace decaf {
       argnum += 1;
     }
   }
+  Context::Context(const Context& ctx, z3::solver&& solver) :
+    Context(ctx)
+  {
+    solver = std::move(solver);
```

Comment:
Can this be done in the MIL?

Reply:
Nope. The nested constructor call counts as initializing all the members and you can't initialize s field twice in the MIL.

I'd rather future-proof things here so I think the nested constructor call is best.